packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -2164,7 +2164,7 @@ class _InputDecoratorState extends State<InputDecorator> with TickerProviderStat
       child: Text(
         decoration!.hintText!,
         style: hintStyle,
-        overflow: TextOverflow.ellipsis,
+        overflow: decoration!.hintOverflow,
```

Comment:
IIRC, we try not to define the default values in the themes themselves, but rather have a fallback mechanism as follows:
```
decoration.hintOverflow > themeData.inputDecorationTheme.hintOverflow > default hintOverflow value,
```

So, then the right thing following that pattern would be:
```suggestion
        overflow: decoration?.hintOverflow ?? themeData.inputDecorationTheme.hintOverflow ?? TextOverflow.ellipsis,
```

Reply:
cc/ @justinmc in case there's something special about `InputDecorator` theming that I might be missing, since the pattern here seems a little different from what I'm used to looking at the surrounding code